### PR TITLE
Update all dependencies

### LINKED ISSUE #340: wait-on 4.4.0
According to https://github.com/smooth-code/jest-puppeteer/blob/master/packages/jest-dev-server/package.json version `4.4.0` contains `wait-on ^4.0.0`. How is it that when I install jest-dev-server my package.lock looks like this?

```json
    "jest-dev-server": {
      "version": "4.4.0",
      "resolved": "https://arm.sero.gic.ericsson.se/artifactory/api/npm/npm-remote/jest-dev-server/-/jest-dev-server-4.4.0.tgz",
      "integrity": "sha1-VXET+q4od0UhYmlqqUweREkasBE=",
      "dev": true,
      "requires": {
        "chalk": "^3.0.0",
        "cwd": "^0.10.0",
        "find-process": "^1.4.3",
        "prompts": "^2.3.0",
        "spawnd": "^4.4.0",
        "tree-kill": "^1.2.2",
        "wait-on": "^3.3.0"
      }
    },
```

Notice `^3.3.0` and not `^4.0.0`?

### LINKED ISSUE #358: jest-dev-server deps difference between repository and npm (wait-on package)
Hi!

In repo jest-dev-server has in deps "wait-on": "^4.0.0" (https://github.com/smooth-code/jest-puppeteer/blob/master/packages/jest-dev-server/package.json)

But when I install jest-dev-server from npm I get in deps "wait-on": "^3.3.0"

How could this happen? And will it be fixed?

UPD: "wait-on": "^4.0.0" still used `request` package as deps. `request` is deprecated now. It would be cool to upgrade `wait-on` to 5 version.

### LINKED ISSUE #360: Increase peerDependencies to allow Puppeteer v4
Also see old issue for v2:

https://github.com/smooth-code/jest-puppeteer/pull/289

### LINKED ISSUE #390: Puppeteer peer dependency outdated
## 🐛 Bug Report

The package peer dependency does not include puppeteer v3, despite multiple issues and people saying they use v3 (stably, I might add). With the advent of npm v7 which hard enforces peer dependencies, it would be great if this could be updated.

```
  "peerDependencies": {
    "puppeteer": ">= 1.5.0 < 3"
  },
 ```

## To Reproduce

1. Install `puppeteer@3.3.0` and `jest-puppeteer@latest`
2. Run `npm update --verbose`
3. Behold a resolve error.

## Expected behavior

1. Install `puppeteer@3.3.0` and `jest-puppeteer@latest`
2. Run `npm update --verbose`
3. Successfully update your packages and move on with your day.

```bash
## System:
 - OS: Linux 5.4 Ubuntu 20.04.2 LTS (Focal Fossa)
 - CPU: (8) x64 Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz
 - Memory: 18.49 GB / 24.98 GB
 - Container: Yes
 - Shell: 5.0.17 - /bin/bash
## Binaries:
 - Node: 10.19.0 - /usr/bin/node
 - npm: 7.6.2 - /usr/local/bin/npm
## npmPackages:
 - jest-puppeteer: ^4.4.0 => 4.4.0
 - puppeteer: ^3.3.0 => 3.3.0
```

### PR DESCRIPTION
**Pull request by UziTech**
## Summary

Update all deps

## Test plan

ran `yarn test`

closes #391
closes #389
closes #384
closes #383
closes #376 
closes #375 
closes #374 
closes #348
closes #345

fixes #390
fixes #360 
fixes #358 
fixes #340

### CODE REVIEW

File: packages/jest-dev-server/package.json [hunk 1]
```diff
@@ -12,18 +12,18 @@
...
+    "chalk": "^4.1.0",
     "cwd": "^0.10.0",
-    "find-process": "^1.4.3",
-    "prompts": "^2.3.0",
+    "find-process": "^1.4.4",
+    "prompts": "^2.4.1",
     "spawnd": "^4.4.0",
     "tree-kill": "^1.2.2",
-    "wait-on": "^4.0.0"
+    "wait-on": "^5.3.0"
```
Comment (UziTech) [inline packages/jest-dev-server/package.json:27]: This is most likely a breaking change since `wait-on` uses `axios` instead of `request` in v5

**Review by gregberge — APPROVED** [review]

**Comment by gaaliche** [comment]
Great work @UziTech 👏🏼

**Comment by sergtimosh** [comment]
@UziTech Thanks for capturing the flag!

**Comment by gaaliche** [comment]
@UziTech when will the new release be published?

**Comment by UziTech** [comment]
Soon. This will be a major update so I wanted to get a few more breaking changes in before releasing.

**Pull request merged**